src/hitable/Sky.cpp

```diff
@@ -10,30 +10,29 @@ using namespace CGL;
 using namespace std;
 using namespace cv;
 
-Sky::Sky(Mat texture, double radius) {
-    this->radius = radius;
+Sky::Sky(Mat texture, double radius) : textureMap(texture.cols, texture.rows) {
+    radius = radius;
     radiusSqr = radius * radius;
-    if (texture != NULL) {
-        textureMap = new SphericalMapping(texture.cols, texture.rows);
+    if (!texture.empty()) {
```

Comment:
`texture.empty()` checks 2 cases -- i.e. a null check and a zero matrix check
I'm not sure we want the zero matrix check here

Reply:
I'm not entirely sure either but you can't do `texture != null` without a compile error.
I assume a zero matrix is textureMap initialized with nothing. 
Either way we should keep this in mind in case it causes problems later.

Reply:
Maybe try using `M.data == NULL`